### PR TITLE
Allow setting the Glue STS endpoint and region

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

Add config properties to specify the STS service to use when authenticating to AWS Glue. This is needed for situations like deploying in GovCloud where the global STS endpoint should not be used.

## Non-technical explanation

Add options which allow for Glue to be used in situations like GovCloud.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Allow setting the AWS STS endpoint and region when using a Glue metastore.

# Delta Lake
* Allow setting the AWS STS endpoint and region when using a Glue metastore.

# Iceberg
* Allow setting the AWS STS endpoint and region when using a Glue catalog.
```

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueCredentialsProvider.java [hunk 1]
```diff
@@ -45,10 +48,24 @@ public GlueCredentialsProvider(GlueHiveMetastoreConfig config)
...
+                    stsClientBuilder.setRegion(config.getGlueStsRegion().get());
+                }
+                else if (config.getPinGlueClientToCurrentRegion()) {
+                    stsClientBuilder.setRegion(getCurrentRegionFromEC2Metadata().getName());
+                }
+
                 provider = new STSAssumeRoleSessionCredentialsProvider
                         .Builder(config.getIamRole().get(), "trino-session")
                         .withExternalId(config.getExternalId().orElse(null))
-                        .withLongLivedCredentialsProvider(provider)
```
Comment (alexjo2144) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueCredentialsProvider.java:51]: Note for reviewers: `withLongLivedCredentialsProvider` is deprecated and cannot be used along with `withStsClient`, that's why this change is here.

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -606,6 +606,13 @@ Property Name                                        Description
 ``hive.metastore.glue.endpoint-url``                 Glue API endpoint URL (optional).
                                                      Example: ``https://glue.us-east-1.amazonaws.com``
 
+``hive.metastore.glue.sts.region``                   AWS region of the STS service to authenticate with. This is
```
Comment (findepi) [inline docs/src/main/sphinx/connector/hive.rst:609]: Do we need a new setting? why this cannot be based on `hive.metastore.glue.region`?

Comment (alexjo2144) [reply docs/src/main/sphinx/connector/hive.rst:609]: It is necessary if we want to support situations like using Minio's STS with AWS Glue. 

There's also this: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html

TLDR: By default STS uses the global endpoint which is always available. You can set a region explicitly to reduce latency but STS is not available in all regions. So if we just use the Glue region but you're deployed in one of those places, for example `eu-south-1`, you need to use either the global endpoint or a diferent region with STS available.

I believe this means we need the setting.

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
`ci / build-pt (pull_request)` failed (https://github.com/trinodb/trino/issues/14453)
tests need to be re-run
